dev/NavigationView/NavigationViewAutomationPeer.cpp

```diff
@@ -50,3 +50,18 @@ winrt::com_array<winrt::Windows::UI::Xaml::Automation::Provider::IRawElementProv
     }
     return {};
 }
+
+void NavigationViewAutomationPeer::RaiseSelectionChangedEvent(winrt::IInspectable const& oldSelection, winrt::IInspectable const& newSelecttion)
+{
+    if (winrt::AutomationPeer::ListenerExists(winrt::AutomationEvents::SelectionPatternOnInvalidated))
+    {
+        //RaiseAutomationEvent(winrt::AutomationEvents::SelectionPatternOnInvalidated);
+        //RaiseAutomationEvent(winrt::AutomationEvents::SelectionItemPatternOnElementSelected);
+
+        // box_value(oldState) doesn't work here, use ReferenceWithABIRuntimeClassName to make Narrator can unbox it.
+        //RaisePropertyChangedEvent(winrt::SelectionPatternIdentifiers::SelectionProperty(),
+        //    box_value(oldSelection),
+        //    box_value(newSelecttion)
+        //);
```

Comment:
Do we have theories about why this isn't working?

Reply:
Other properties seems to update just fine, the `SelectionPattern.SelectionProperty()` however didn't seem to work. Maybe it could be that we should be returning a specific type here?

The problem is that I haven't found much information on the `SelectionPatter.SelectionProperty` property or on how to use it.

Reply:
@ojhad thoughts on this?

Reply:
@chingucoding I believe that you have to raise `SelectionItemPatternOnElementSelected` on the peer of the selected item itself.

So, for example, if you replace the commented out code you have here with the following, you should get the intended result:

```
if (auto nv = Owner().try_as<winrt::NavigationView>())
{
        if (auto nvi = winrt::get_self<NavigationView>(nv)->GetSelectedContainer())
        {
            if (auto peer = winrt::FrameworkElementAutomationPeer::CreatePeerForElement(nvi))
            {
                peer.RaiseAutomationEvent(winrt::AutomationEvents::SelectionItemPatternOnElementSelected);
            }
        }
}
```

Reply:
Thanks for the fix @ojhad !

I was able to fix the change being announced twice through quite bad hack. The issue is that expand/collapse get's changed after selection and is not aware of latest selection changes. That's why we suppress raising selection changed event iff an item has children and will be selected. If an item get's expanded but not selected we raise expand/collapse change and if an item changes selection but has no children, we raise selection changed.